File: src/main/kotlin/org/javagirls/capitulo_2/java/StringTemplate.java
package org.javagirls.capitulo_2.java;

import java.util.Scanner;

public class StringTemplate {
    public static void main(String[] args) {
        System.out.println("JAVA --------------------------");
        Scanner scanner = new Scanner(System.in);
        String name = scanner.nextLine();
        // concatenação básica no Java
        System.out.println("Hello, " + name + "!");
        // semelhante ao string template em Kotlin
        System.out.println(String.format("Hello, %s!", name));
    }
}

File: src/main/kotlin/org/javagirls/capitulo_2/StringTemplate.kt
package org.javagirls.capitulo_2

fun main() {
    println("KOTLIN --------------------------")
    val input = readln()
    val name = if (input.isNotBlank()) input else "Kotlin"

    println("Hello, $name!")

    // complex expression dentro de um string template, utilizar chaves
    println("Hello, ${name.length} letter person!")

    // utilizar barra invertida para imprimir $
    println("\$ utilizar barra invertida para imprimir $")

    // complex expression utilizando chaves ( pode conter aspas )
    println("Hello, ${if (name.isBlank()) "someone" else name }!")

}